core/app/assets/javascripts/spree.js.erb

```diff
@@ -51,13 +51,45 @@ Spree.ajax = function(url, options) {
   return $.ajax(url, options);
 };
 
-Spree.routes = {
-  states_search: Spree.pathFor('api/states'),
-  apply_coupon_code: function(order_id) {
-    return Spree.pathFor("api/orders/" + order_id + "/coupon_codes");
+/**
+ * @deprecated Spree.routes will be removed in a future release. Please use Spree.pathFor instead.
+ * See: https://github.com/solidusio/solidus/issues/3405
+ */
+Spree.routesDeprecationProxy = {
+  get: function(obj, prop) {
+    <% if Rails.env != "production" %>
+      console.log("Spree.routes is deprecated, please use pathFor instead. See: https://github.com/solidusio/solidus/issues/3405");
+    <% end %>
```

Comment:
Would it be better to actually pass the environment to the JS code at runtime? I'm not a huge fan of mixing Ruby and JS like this, I think it could potentially generate a lot of confusion and subtle issues.

Reply:
I'm not quite sure how to go about this - do you have an example? We could assign a JS variable at the top of the file `var production = <%= Rails.env == "production" %>`, is that what you mean?

Also, I noticed in the frontend JS file, there's another deprecation done like this:
```
  if (console && console.warn) {
    console.warn('Spree.url is deprecated, and will be removed from a future Solidus version.');
  }
```
I hadn't noticed it before - would it be better to follow the already existing example set there? 

Thanks!

Reply:
@seand7565 100%, I didn't realize we had done it before. This will print the warning even in production, but IMO it's okay to be consistent and follow the existing pattern. If we want to only print the warnings in production, let's make sure we change it everywhere, and maybe in a separate PR.

@kennyadsl what do you think?

Reply:
Considering it only happens in the admin, I think it's not too bad if we print a deprecation warning in the console. So 👍 on making them more uniform. 

For the other PR to use the `Rais.env` check, I guess the JS file in the frontend is not an `.erb` and I wouldn't add ruby to it since people can be including frontend files in unexpected ways (maybe with webpack?) and this would be a breaking change.